dm/dm/master/scheduler/scheduler.go

```diff
@@ -1052,10 +1052,37 @@ func (s *Scheduler) StartRelay(source string, workers []string) error {
 	}
 
 	// 1. precheck
-	if _, ok := s.sourceCfgs[source]; !ok {
+	sourceCfg, ok := s.sourceCfgs[source]
+	if !ok {
 		return terror.ErrSchedulerSourceCfgNotExist.Generate(source)
 	}
 	startedWorkers := s.relayWorkers[source]
+
+	// quick path for `start-relay` without worker name
+	if len(workers) == 0 {
+		if len(startedWorkers) != 0 {
+			return terror.ErrSchedulerStartRelayOnSpecified.Generate(startedWorkers)
+		}
+		// update enable-relay in source config
+		sourceCfg.EnableRelay = true
```

Comment:
Can we return here if enable relay already be true?

Reply:
OK, I think we can skip adjust source config and still notify worker by source bound when worker is somehow not received the first start-relay command

Reply:
If so, I think no need to address this comment.